File: HW/hw8/HW8.py
'''
Created on Feb 28, 2024

@author: alexxavier
'''

class HW6:
    def longestPalindrome(self, s):
        char_counts = {}
        total = 0
        largest_odd = 0

        for char in s:
            if char not in char_counts:
                char_counts[char] = 1
            else:
                char_counts[char] += 1

        for count in char_counts.values():
            if count % 2 == 0:
                total += count

            if count % 2 != 0 and count > largest_odd:
                largest_odd = count

        return total + largest_odd
    
    

hw6 = HW6()
result = hw6.longestPalindrome("abccccdd")
print(result)

File: HW/hw8/HW8.java
package HW;

import java.util.HashMap;

public class HW8 {

	public int longestPalindrome(String s) {
		char[] charArr = s.toCharArray();
		HashMap<Character, Integer> charCounts = new HashMap<Character, Integer>();
		int total = 0;
		int largestOdd = 0;

		for (int i = 0; i < charArr.length; i++) {
			if (!charCounts.containsKey(charArr[i])) {
				charCounts.put(charArr[i], 1);
			} else {
				charCounts.put(charArr[i], charCounts.get(charArr[i]) + 1);

			}

		}

		for (Character c : charCounts.keySet()) {
			if ((charCounts.get(c) % 2) == 0) {
				total += charCounts.get(c);
			}

			if ((charCounts.get(c) % 2) != 0 && charCounts.get(c) > largestOdd) {
				largestOdd = charCounts.get(c);

			}

		}
		return total + largestOdd;

	}

	public static void main(String[] args) {
		HW8 x = new HW8();
		System.out.println(x.longestPalindrome("abccccdd"));
	}

}
